internal/pkg/cli/deploy/env.go

```diff
@@ -67,8 +59,16 @@ func (d *envDeployer) UploadArtifacts() (map[string]string, error) {
 	return urls, nil
 }
 
+// DeployEnvironmentInput contains information used to deploy the environment.
+type DeployEnvironmentInput struct {
+	RootUserARN         string
+	IsProduction        bool
```

Comment:
Are we keeping this in case users have existing envs that they inited with the now-deprecated `--prod` flag?

Reply:
The flag was deprecated in `cli` package but all these wiring behind `cli` wasn't removed (for example, `deploy.CreateEnvironmentInput` still has the field `Prod`). Therefore I thought we wanted to keep the wirings there so that we can revive the flag in the future when we want. This is to keep the existing wirings.

Reply:
I _think_ we've moved away from a prescribed `prod` setting in favor of giving users a more fine-tuned config-setting experience, but I can't speak for the team!

Reply:
Oh, [this](https://github.com/aws/copilot-cli/issues/1851) has an idea for that flag for the future.